service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ActionCMOController.java

```diff
@@ -0,0 +1,146 @@
+                               UploadDocumentService uploadDocumentService) {
+        this.draftCMOService = draftCMOService;
+        this.caseManageOrderActionService = caseManageOrderActionService;
+        this.docmosisDocumentGeneratorService = docmosisDocumentGeneratorService;
+        this.uploadDocumentService = uploadDocumentService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> caseDataMap = caseDetails.getData();
+
+        draftCMOService.prepareCustomDirections(caseDetails.getData());
+
+        CaseManagementOrder caseManagementOrderToBeActioned =
+            caseManageOrderActionService.getCaseManagementOrderForAction(caseDataMap);
+        caseDetails.getData().put(CASE_MANAGEMENT_ORDER_KEY, caseManagementOrderToBeActioned);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) throws IOException {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> caseDataMap = caseDetails.getData();
+
+        CaseManagementOrder caseManagementOrderToBeActioned =
+            caseManageOrderActionService.getCaseManagementOrderForAction(caseDataMap);
+
+        Document updatedDraftCMODocument = getDocument(authorization, userId, caseDataMap, false);
+        CaseManagementOrder updatedDraftCaseManagementOrderWithDocument =
+            caseManageOrderActionService.addDocumentToCaseManagementOrder(caseManagementOrderToBeActioned,
+                updatedDraftCMODocument);
+
+        caseDetails.getData().put(CASE_MANAGEMENT_ORDER_ACTION_KEY, ImmutableMap.of("orderDoc",
+            updatedDraftCaseManagementOrderWithDocument.getOrderDoc()));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) throws IOException {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+
+        CaseManagementOrder updatedDraftCaseManagementOrder =
+            caseManageOrderActionService.getCaseManagementOrderForAction(caseDetails.getData());
+
+        boolean judgeApprovedDraftCMO = hasJudgeApprovedDraftCMO(updatedDraftCaseManagementOrder);
```

Comment:
not sure this needs to be abstracted to it's own variable

Reply:
I would also rather hasJudgeApproved method name. Context is given by what you pass into the method